src/Razor/src/Microsoft.VisualStudio.RazorExtension/AssemblyBindingRedirects.cs

```diff
@@ -38,3 +38,17 @@
     OldVersionLowerBound = "0.0.0.0",
     OldVersionUpperBound = "5.0.0.0",
     NewVersion = "5.0.0.0")]
+// [assembly: ProvideBindingRedirection(
+//     AssemblyName = "MediatR",
+//     GenerateCodeBase = true,
+//     PublicKeyToken = "bb9a41a5e8aaa7e2",
+//     OldVersionLowerBound = "8.0.0.0",
+//     OldVersionUpperBound = "8.1.0.0",
+//     NewVersion = "8.1.0.0")]
+[assembly: ProvideBindingRedirection(
```

Comment:
This definitely isn't ideal. @david-driscoll is this requirement introduced based on a dependency that O# depends on outside of our control?

Reply:
I think I've bumped all of Msft.Ext to `3.1`

I'm not using any new features so technically they can drop back down to 2.1 if you need.

Reply:
I think that would be really helpful for us, otherwise we need this redirect and it might cause big problems.

Reply:
Is this something we can get access to early? Only asking because it's gonna be an ngen pain to insert this 😄

Reply:
Does this apply to logging as well, or just DI?

Currently we have these:

```xml
    <PackageReference Update="Microsoft.Extensions.Logging" Version="3.1.0" />
    <PackageReference Update="Microsoft.Extensions.Logging.Abstractions" Version="3.1.0" />
    <PackageReference Update="Microsoft.Extensions.Logging.Debug" Version="3.1.0" />
    <PackageReference Update="Microsoft.Extensions.Configuration.Abstractions" Version="3.1.0" />
    <PackageReference Update="Microsoft.Extensions.Configuration" Version="3.1.0" />
    <PackageReference Update="Microsoft.Extensions.DependencyInjection" Version="3.1.0" />
    <PackageReference Update="Microsoft.Extensions.DependencyInjection.Abstractions" Version="3.1.0" />
```

I can drop all or some of them.  down to 2.1 or 2.0?

Reply:
So we don't necessarily care about which version they are as long as the dependency graph is coherent. Given we run our language server in VS which is a .NET framework app it relies on a hugeeee set of binding redirects to ensure that every dependency uses the correct version of a dll; however, every time we have to include a binding redirect we essentially restrict what versions of an assembly can exist for other teams because we then binding redirect it to our own. Hope that makes sense 😄

Reply:
If you hit me up on Teams or Slack I can figure out the best solution.

Reply:
If this works for you guys, heres the proposed change: https://github.com/OmniSharp/csharp-language-server-protocol/pull/342